dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -149,36 +121,33 @@ object Datadog {
     @JvmStatic
     @JvmOverloads
     fun getInstance(instanceName: String? = null): SdkCore? {
-        return registry.getInstance(instanceName)
+        return synchronized(registry) {
+            registry.getInstance(instanceName)
+        }
     }
 
-    // endregion
-
-    // region Global methods
-
     /**
-     * Clears all data that has not already been sent to Datadog servers.
+     * Stop the initialized SDK instance attached to the given name,
+     * or the default instance if the name is null.
+     * @param instanceName the name of the instance to stop,
+     * or null to stop the default instance
      */
     @JvmStatic
-    @Deprecated(
-        "RUMM-3103 use getInstance().clearAllData()",
-        ReplaceWith(
-            "Datadog.getInstance()?.clearAllData()"
-        )
-    )
-    fun clearAllData() {
-        // TODO RUMM-3103 deprecate this
-        globalSdkCore.clearAllData()
+    @JvmOverloads
+    fun stopInstance(instanceName: String? = null) {
+        synchronized(registry) {
+            val instance = registry.unregister(instanceName)
+            instance?.stop()
+        }
     }
 
+    // endregion
+
+    // region Global methods
+
     // Stop all Datadog work (for test purposes).
     internal fun stop() {
```

Comment:
seems like this method can be removed, given that we can call `stopInstance` directly from public API.